packages/@aws-cdk/aws-fsx/README.md

```diff
@@ -17,8 +17,188 @@
+video processing, and financial modeling.
+
+The open-source Lustre file system is designed for applications that require fast storage—where you want your storage
+to keep up with your compute. Lustre was built to solve the problem of quickly and cheaply processing the world's
+ever-growing datasets. It's a widely used file system designed for the fastest computers in the world. It provides
+submillisecond latencies, up to hundreds of GBps of throughput, and up to millions of IOPS. For more information on
+Lustre, see the [Lustre website](http://lustre.org/).
+
+As a fully managed service, Amazon FSx makes it easier for you to use Lustre for workloads where storage speed matters.
+Amazon FSx for Lustre eliminates the traditional complexity of setting up and managing Lustre file systems, enabling
+you to spin up and run a battle-tested high-performance file system in minutes. It also provides multiple deployment
+options so you can optimize cost for your needs.
+
+Amazon FSx for Lustre is POSIX-compliant, so you can use your current Linux-based applications without having to make
+any changes. Amazon FSx for Lustre provides a native file system interface and works as any file system does with your
+Linux operating system. It also provides read-after-write consistency and supports file locking.
+
+### Installation
+
+Import to your project:
 
 ```ts
 import fsx = require('@aws-cdk/aws-fsx');
 ```
+
+### Basic Usage
+
+Setup required properties and create:
+
+```ts
+const stack = new Stack(app, 'Stack');
+const vpc = new Vpc(stack, 'VPC');
+const storageCapacity = 1200;
+const lustreConfiguration = {
+  deploymentType: FsxLustreDeploymentType.SCRATCH_2
+};
+
+const fileSystem = new FsxLustreFileSystem(stack, 'FsxLustreFileSystem', {
+  lustreConfiguration,
+  storageCapacity,
+  vpc,
+  vpcSubnet: vpc.privateSubnets[0]});
+```
+
+### Connecting
+
+To control who can access the file system, use the `.connections` attribute. FSx has a fixed default port, so you don't
+need to specify the port. This example allows an EC2 instance to connect to a file system:
+
+```ts
+fileSystem.connections.allowDefaultPortFrom(instance);
+```
+
+### Mounting
+
+To mount an FSx for Lustre file system programmatically, the mount name of the file system needs to be determined. The
+easiest way to do that is with the AWS CLI, which can be written into a User Data script to be run on start-up of your
+instance. The following example can be used for reference on how to bring up a file system and EC2 instance with the
+file system mounted on the instance at start-up:
+
+```ts
+const app = new App();
+const stack = new Stack(app, 'AwsCdkFsxLustre');
+const vpc = new Vpc(stack, 'VPC');
+
+// Create a security group to share across the file system and ec2 instance and open the FSx ports.
+const securityGroup = new SecurityGroup(stack, 'FsxLustreSecurityGroup', {
```

Comment:
You seem to be going against your own advice of the previous section here. Quote:

> FSx has a fixed default port, so you don't need to specify the port.

Also, generally you wouldn't share an SG across instance and filesystem. As soon as you attach your instance to a load balancer, you would implicitly open port 8000 (or whatever) on the filesystem as well.

Reply:
This example is rather convoluted due to the way the EC2 instance needs to look the file system mount name up for itself.

The problem I was trying to avoid here was having the EC2 instance come up before the file system is up and the mount name is retrievable from the AWS CLI. To do that, I added the dependency on the file system from the instance. If I let both resources create their own security groups and tried to add this dependency, it would be circular, as the file system already attempts to add the instance's security group to its Connections.

Removing this dependency would simplify the example a fair bit, but there's always the chance the instance runs its UserData before the file system is initialized and the mount fails.

Reply:
Oh I see. The security groups are accidentally included in the dependency set when you do `addDependency` huh?

Let's replace it with:

```
// Depend on the filesystem resource, not any of the security groups and security group rules
instance.addDependency(fileSystem.node.defaultChild);
```

Which I agree is not great, but that's more of a symptom of the dependency system. We can address that in due time, but that's no reason to make this example substandard.

Reply:
This worked.